examples/msgpack/src/main.rs

```diff
@@ -0,0 +1,39 @@
+#![feature(plugin)]
+#![plugin(rocket_codegen)]
+
+extern crate rocket;
+extern crate rocket_contrib;
+#[macro_use] extern crate serde_derive;
+
+#[cfg(test)] mod tests;
+
+use rocket_contrib::MsgPack;
+
+#[derive(Serialize, Deserialize)]
+struct Message {
+    id: usize,
+    contents: String
+}
+
+#[get("/<id>", format = "application/msgpack")]
+fn get(id: usize) -> MsgPack<Message> {
+    MsgPack(Message {
+        id: id,
+        contents: "Hello, world!".to_string(),
+    })
+}
+
+#[post("/", data = "<data>", format = "application/msgpack")]
+fn create(data: MsgPack<Message>) -> Result<(), ()> {
```

Comment:
This isn't testable. Just return the contents directly (so return type of this function is `String`) so you can test for them later:

```
data.into_inner().contents
```